src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.Tests.Common.cs

```diff
@@ -154,6 +154,22 @@ public static async Task<Regex[]> GetRegexesAsync(RegexEngine engine, params (st
             RegexEngine.NonBacktrackingSourceGenerated => RegexOptionNonBacktracking | RegexOptions.Compiled,
             _ => throw new ArgumentException($"Unknown engine: {engine}"),
         };
+
+        /// <summary>Set the AppContext variable REGEX_NONBACKTRACKING_MAX_AUTOMATA_SIZE to the given max value. Only used with Nonbacktracking engine.</summary>
+        public static void SetSafeSizeThreshold(int maxSize)
+        {
+#if NET7_0_OR_GREATER
+            AppContext.SetData("REGEX_NONBACKTRACKING_MAX_AUTOMATA_SIZE", maxSize);
+#endif
```

Comment:
Are these ifdefs actually needed?
```suggestion
            AppContext.SetData("REGEX_NONBACKTRACKING_MAX_AUTOMATA_SIZE", maxSize);
```

Reply:
They are because `AppContext.SetData` is not supported in earlier versions, it will not compile otherwise.

Reply:
Ah, SetData is new.  I see.  GetData has been around forever.